File: frontend/lib/models/answer.dart
class Answer {
  final String id;
  final String reply;
  final String uid;
  final String datetime;
  final String qid;

  const Answer(
      {required this.id,
      required this.reply,
      required this.uid,
      required this.datetime,
      required this.qid});

  factory Answer.fromJson(Map<String, dynamic> json) {
    return Answer(
        id: json['id'],
        reply: json['reply'],
        uid: json['uid'],
        datetime: json['datetime'],
        qid: json['qid']);
  }
  static List<Answer> fromListJson(List<dynamic> json) {
    List<Answer> result = <Answer>[];
    for (Map<String, dynamic> d in json) {
      result.add(Answer.fromJson(d));
    }
    return result;
  }
}

File: aspnet/Model/answer.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
namespace aspnet.models;


public class Answer
{
    [BsonId][BsonRepresentation(BsonType.ObjectId)]  
    public string? id { get; set; }
    [BsonElement]
    public string reply { get; set; }
    [BsonElement]
    public string uid { get; set; }
    [BsonElement]
    public string datetime { get; set; }
    [BsonElement]
    public string qid { get; set; }

    public Answer(string Id, string Reply, string Uid, string Datetime, string Qid)
    {
        this.id = Id;
        this.reply = Reply;
        this.uid = Uid;
        this.datetime = Datetime;
        this.qid = Qid;
    }
}


